edition = "2024";

package legion.messages;


message PowerProfile
{
    enum Profiles {
        POWER_PROFILE_UNKNOWN      = 0  ;
        POWER_PROFILE_QUIET        = 1  ;
        POWER_PROFILE_BALANCED     = 2  ;
        POWER_PROFILE_PERFORMANCE  = 3  ;
        POWER_PROFILE_EXTREME      = 224;
        POWER_PROFILE_CUSTOM       = 255;
    }

    message CustomFnq {
        bool       current_value  = 1;
        bool       default_value  = 2;
        bool       supported      = 3;
    }

    Profiles   current_value             = 1;  //smart_fan
    Profiles   thermal_mode              = 2;  //thermal mode
    CustomFnq  custom_fnq_enabled        = 3;  //custom fn+q
    repeated Profiles supported_profiles = 4;  //supported profiles
}
